### PR TITLE
Add logs to make it more apparent that No Common Ancestors isn't necessarily bad

### LINKED ISSUE #3337: No common ancestor warnings when doing IBD, but only on tesnet3
Firing up my testnet3 when testing #3227 on  7d89844575f6865765a1da3a34de2b158442774f . This does NOT seem to happen on mainnet for some reason.

and i get a bunch of these warnings when doing IBD

>[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(00000000b643ed6153046ba473f0517ebe5c0d803641ac77736a29eada6d9ee0) prevHash=DoubleSha256BDigestBE(0000000000000024fd615bda2482048a5e150d3896a42ff640b8cbf919cbf428)
[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(00000000000000334d1854ab0ad10dcd91caf576e57027e3d82b7d3cf0e666e4) prevHash=DoubleSha256BDigestBE(00000000b643ed6153046ba473f0517ebe5c0d803641ac77736a29eada6d9ee0)
[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(000000000000002b6617304093a55e22a68a92b25f1c3eaa129165c2261a8bd3) prevHash=DoubleSha256BDigestBE(00000000000000334d1854ab0ad10dcd91caf576e57027e3d82b7d3cf0e666e4)
[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(0000000054dce4ff9a8ca12e12c702d6017a6a14f59d65e4ab2bef603d1d761c) prevHash=DoubleSha256BDigestBE(000000000000002b6617304093a55e22a68a92b25f1c3eaa129165c2261a8bd3)
[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(00000000e3d897f39cb09b560a8fac31a2d7e1f2860cd7de1815e1500d33e2c9) prevHash=DoubleSha256BDigestBE(0000000054dce4ff9a8ca12e12c702d6017a6a14f59d65e4ab2bef603d1d761c)
[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(000000000000002a776461bdcdbec9e10175f0d3d2ef9236dae85f51ffd132a5) prevHash=DoubleSha256BDigestBE(00000000e3d897f39cb09b560a8fac31a2d7e1f2860cd7de1815e1500d33e2c9)
[info] 2021-06-28T12:28:23UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=DoubleSha256BDigestBE(000000009f2db2728830336e19e16b43c7669b8c394300bb48d89c5e4767ab25) prevHash=DoubleSha256BDigestBE(000000000000002a776461bdcdbec9e10175f0d3d2ef9236dae85f51ffd132a5)
[info] 2021-06-28T12:28:28UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
[info] 2021-06-28T12:28:29UTC INFO [DataMessageHandler] Got 659 compact filter header hashes
[info] 2021-06-28T12:28:32UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters in datamessagehandler
[info] 2021-06-28T12:28:32UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=2006058, stopBlockHash=000000009f2db2728830336e19e16b43c7669b8c394300bb48d89c5e4767ab25)

### PR DESCRIPTION
**Pull request by Christewart**
fixes #3337 

TLDR, there is no bugs here. Just misleading log messages. What was happening on my node was we have a competing chain split in the blockchain. On the tip that is dying, everytime we get a new block header we log this `warn` message

>[info] 2021-06-29T19:21:09UTC WARN [Blockchain$] No common ancestor found in the chain with tip=0000000069833246687fdb88a73b55e7cf1694f10c47d892ad00d90173234c6c to connect to hash=000000008576169f92b556ba1662530a2876287704da31e511145354bcee8fbc prevHash=000000000000003be5dab5ccebec3da4c29795d1378f14efae1f5e4ced4d6e17

HOWEVER, we silently succeed connecting the tip to the _other_ side of the chainsplit behind the scenes. 

This PR modifies the `No common ancestor` log message to indicate that this might be on the wrong side of a chain split.

Right now, i've modified this log message 

>[info] 2021-06-30T13:50:50UTC INFO [Blockchain$] Successfully verified=000000000000000a020b9d3be275137d35a53950e511e245f1ae9c701dbfb851, connecting to chain

to be `INFO` rather than `DEBUG` like it was before. The downside of this approach though is during IBD, we will get a _ton_ of those log messages. I'm open to moving it back to `DEBUG`, but would like opinions from others first.

### CODE REVIEW

**Comment by benthecarman** [comment]
imo, that log should stay as `debug`, maybe the best solution would be to make the `No common ancestor` log as `debug` as well, and then add a new `warn` log that only is done if all of the calls to `connectTip` fail.

**Comment by Christewart** [comment]
> imo, that log should stay as `debug`, maybe the best solution would be to make the `No common ancestor` log as `debug` as well, and then add a new `warn` log that only is done if all of the calls to `connectTip` fail.

This seems like a really good idea.

**Comment by Christewart** [comment]
> > imo, that log should stay as `debug`, maybe the best solution would be to make the `No common ancestor` log as `debug` as well, and then add a new `warn` log that only is done if all of the calls to `connectTip` fail.
> 
> This seems like a really good idea.

So it looks like this is actually already handled here. We return a failed future if we cannot connect to any internal blockchains.
https://github.com/bitcoin-s/bitcoin-s/blob/7ba7f8b9ba5ed331728183a067b2a45c6cfcfeb8/chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala#L142

**Review by benthecarman — APPROVED** [review]
